src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/FilingService.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.companieshouse.confirmationstatementapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.confirmationstatementapi.exception.SubmissionNotFoundException;
+import uk.gov.companieshouse.confirmationstatementapi.model.json.ConfirmationStatementSubmissionJson;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class FilingService {
+
+    @Value("${CONFIRMATION_STATEMENT_NO_UPDATES}")
+    private String filingDescription;
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+    private ConfirmationStatementService confirmationStatementService;
+
+    @Autowired
+    public FilingService(ConfirmationStatementService csService) {
+        confirmationStatementService = csService;
+    }
+
+    public FilingApi generateConfirmationFiling(String confirmationStatementId) throws SubmissionNotFoundException {
+        FilingApi filing = new FilingApi();
+        filing.setKind("Confirmation Statement");
+        setFilingApiData(filing, confirmationStatementId);
+        setDescription(filing);
+        return filing;
+    }
+
+    private void setFilingApiData(FilingApi filing, String confirmationStatementId) throws SubmissionNotFoundException {
+        Optional<ConfirmationStatementSubmissionJson> submissionOpt =
+                confirmationStatementService.getConfirmationStatement(confirmationStatementId);
+        ConfirmationStatementSubmissionJson submission = submissionOpt
+                .orElseThrow(() ->
+                        new SubmissionNotFoundException(
+                                String.format("Empty submission returned when generating filing for %s", confirmationStatementId)));
+        if (submission.getData() != null) {
+            Map<String, Object> data = new HashMap<>();
+            data.put("confirmationStatementDate", submission.getData().getMadeUpToDate());
+            data.put("tradingOnMarket", !submission.getData().getTradingStatusData().getTradingStatusAnswer());
+            data.put("dtr5Ind", false);
+            filing.setData(data);
+        } else {
+            throw new SubmissionNotFoundException(
+                    String.format("Submission contains no data %s", confirmationStatementId));
+        }
+    }
+
+    private void setDescription(FilingApi filing) {
+        LocalDate madeUpToDate = (LocalDate)filing.getData().get("confirmationStatementDate");
```

Comment:
Is there any reason description needs to be in its own method? If it was inside the set data method we wouldnt need to case the date here.

Reply:
The method might be a bit long and harder to read if we put it all together, though yes we do have that as a trade off. I was hoped you'd agree that the price is right.

Reply:
you could call setDescription from within setData, and pass the CS submission in? Keeps the method easy to read, and we wont need to cast anything.